src/openapi-generator.ts

```diff
@@ -397,7 +397,7 @@ export class OpenAPIGenerator {
   ): SchemaObject | ReferenceObject {
     const simpleSchema = this.generateSimpleSchema(zodSchema);
 
-    if (simpleSchema.$ref) {
+    if ('$ref' in simpleSchema) {
```

Comment:
This is not exactly equivalent, maybe `if ('$ref' in simpleSchema && simpleSchema.$ref) {`?

Reply:
Oopsie good catch!